modules/stochastic_tools/src/samplers/AdaptiveImportanceSampler.C

```diff
@@ -0,0 +1,196 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "AdaptiveImportanceSampler.h"
+#include "Distribution.h"
+#include "Normal.h"
+#include "Uniform.h"
+
+registerMooseObjectAliased("StochasticToolsApp", AdaptiveImportanceSampler, "AdaptiveImportance");
+
+InputParameters
+AdaptiveImportanceSampler::validParams()
+{
+  InputParameters params = Sampler::validParams();
+  params.addClassDescription("Adaptive Importance Sampler.");
+  params.addRequiredParam<std::vector<DistributionName>>(
+      "distributions",
+      "The distribution names to be sampled, the number of distributions provided defines the "
+      "number of columns per matrix.");
+  params.addRequiredParam<ReporterName>("inputs_reporter", "Reporter with input parameters.");
+  params.addRequiredParam<std::vector<Real>>("proposal_std",
+                                             "Standard deviations of the proposal distributions");
+  params.addRequiredParam<Real>("output_limit", "Limiting values of the VPPs");
+  params.addRequiredParam<std::vector<Real>>(
+      "initial_values", "Initial input values to get the importance sampler started");
```

Comment:
I'm curious if it would be a better idea to put this parameter in `AdaptiveMonteCarloDecision`.